lib/geo_redirect/middleware.rb

```diff
@@ -61,9 +62,13 @@ def force_redirect?
 
     def skip_redirect?
       url = URI.parse(@request.url)
-      query_includes_skip_geo?(url) ||
-        path_not_whitelisted?(url) ||
-        path_blacklisted?(url)
+      host = host_by_hostname(url.host)
+
+      return false unless query_includes_skip_geo?(url) ||
+                          path_not_whitelisted?(url) ||
+                          path_blacklisted?(url)
+      remember_host(host) if @only_first
```

Comment:
I would not add this side-effect to `skip_redirect?` which is supposed to be a very simple predicate. I would add the logic in an outer layer